Sources/Datasets/datasets/image/MNIST.swift

```diff
@@ -0,0 +1,113 @@
+
+    private let data: Tensor<Float>
+    private let labels: Tensor<Int32>
+
+    private init(split: S5TFSplit, data: Tensor<Float>, labels: Tensor<Int32>, batchSize: Int? = nil) {
+        self.split = split
+        self.data = data
+        self.labels = labels
+        self.batchSize = batchSize
+        self.count = data.shape.dimensions.first!
+    }
+
+    public init(split: S5TFSplit) {
+        let baseURL = "https://storage.googleapis.com/cvdf-datasets/mnist"
+        var dataURL: URL
+        var labelsURL: URL
+
+        switch split {
+            case .train:
+                dataURL = S5TFUtils.downloadAndExtract(remoteURL: URL(string: baseURL + "/train-images-idx3-ubyte.gz")!,
+                                                       cacheName: "mnist", fileName: "train-images.gz")!
+                labelsURL = S5TFUtils.downloadAndExtract(remoteURL: URL(string: baseURL + "/train-labels-idx1-ubyte.gz")!,
+                                                         cacheName: "mnist", fileName: "train-labels.gz")!
+            case .test:
+                dataURL = S5TFUtils.downloadAndExtract(remoteURL: URL(string: baseURL + "/t10k-images-idx3-ubyte.gz")!,
+                                                       cacheName: "mnist", fileName: "test-images.gz")!
+                labelsURL = S5TFUtils.downloadAndExtract(remoteURL: URL(string: baseURL + "/t10k-labels-idx1-ubyte.gz")!,
+                                                         cacheName: "mnist", fileName: "test-labels.gz")!
+            default:
+                fatalError("Split does not exist for this dataset.")
+        }
+
+        let rawData = [UInt8](try! Data(contentsOf: URL(string: "file://" + dataURL.absoluteString)!)).dropFirst(16).map(Float.init)
+        let rawLabels = [UInt8](try! Data(contentsOf: URL(string: "file://" + labelsURL.absoluteString)!)).dropFirst(8).map(Int32.init)
+
+        let dataTensor = Tensor<Float>(shape: [rawData.count, 28 * 28], scalars: rawData) / 255.0
+        let labelsTensor = Tensor<Int32>(rawLabels)
+
+        self.init(split: split, data: dataTensor, labels: labelsTensor)
+    }
+
+    public func batched(_ batchSize: Int) -> MNISTDataLoader {
+        return MNISTDataLoader(split: self.split,
+                               data: self.data,
+                               labels: self.labels,
+                               batchSize: batchSize)
+    }
+
+    public mutating func next() -> S5TFLabeledBatch? {
+        guard let batchSize = batchSize else {
+            fatalError("This data loader does not have a batch size. Set a batch size by calling `.batched(...)`")
+        }
+
+        guard index < (count - 1) else {
+            return nil
+        }
+
+        let thisBatchSize = Swift.min(count - index, batchSize)
+
+        var batchFeatures = [Float]()
+        var batchLabels = [Int32]()
+
+        for i in index ..< (index + thisBatchSize) {
+            batchFeatures.append(data[i].scalar!)
```

Comment:
This is one of the view operators where you should _not_ have  white spaces around the operator. 😉https://google.github.io/swift/#horizontal-whitespace